pkg/handler/collector/git/git_test.go

```diff
@@ -0,0 +1,170 @@
+//
+// Copyright 2022 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package git_collector
+
+import (
+	"context"
+	"io"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"github.com/guacsec/guac/pkg/logging"
+)
+
+func Test_gitCol_RetrieveArtifacts(t *testing.T) {
+	ctx := logging.WithLogger(context.Background())
+	logger := logging.FromContext(ctx)
+	type fields struct {
+		url      string
+		dir      string
+		poll     bool
+		interval time.Duration
+	}
+	tests := []struct {
+		name                   string
+		fields                 fields
+		preCreateDir           bool
+		wantDirEmpty           bool
+		wantErr                bool
+		numberOfFilesCollected int
+	}{{
+		name: "get repo",
+		fields: fields{
+			url:      "https://github.com/shafeeshafee/test-123.git",
+			dir:      "./guac-data-test",
```

Comment:
hey Brandon, can you clarify what would MkTemp's purpose be here? it looks like it serves to create a temp file or directory, which we are already doing for the git collector.

Reply:
It creates it in a `tmpfs` type file system (depending on how the OS sets it up), but tmpfs directories are usually cleaned out once in a while and not part of the calling directory structure.. so in case a test fails or is interrupted before it can end gracefully, we won't have stray files in the code folders, and the OS will help clean them up periodically if we do!